### PR TITLE
Make nuget package support coreCLR

### PR DESCRIPTION
**Pull request by jtattermusch**
See:
https://oren.codes/2015/07/29/targeting-net-core/
https://stackoverflow.com/questions/34136228/unable-to-resolve-assembly-reference-issue-without-frameworkassemblies

Also bumping nuget package version to beta2.

### CODE REVIEW

File: csharp/src/Google.Protobuf/Google.Protobuf.nuspec [hunk 1]
```diff
@@ -14,24 +14,40 @@
...
+        <dependency id="System.Reflection" version="4.0.0" />
+        <dependency id="System.Runtime" version="4.0.0" />
+        <dependency id="System.Runtime.Extensions" version="4.0.0" />
+        <dependency id="System.Text.Encoding" version="4.0.0" />
+        <dependency id="System.Text.RegularExpressions" version="4.0.0" />
+      </group>
+    </dependencies>
   </metadata>
   <files>
     <file src="bin/ReleaseSigned/Google.Protobuf.dll" target="lib/portable-net45+netcore45+wpa81+wp8" />
```
Comment (jskeet) [inline csharp/src/Google.Protobuf/Google.Protobuf.nuspec]: Don't we want to add these files to lib/dotnet as well?

File: csharp/src/Google.Protobuf/Google.Protobuf.nuspec [hunk 2]
```diff
@@ -14,24 +14,40 @@
     <releaseNotes>C# proto3 support</releaseNotes>
     <copyright>Copyright 2015, Google Inc.</copyright>
     <tags>Protocol Buffers Binary Serialization Format Google proto proto3</tags>
+    <dependencies>
```
Comment (jskeet) [inline csharp/src/Google.Protobuf/Google.Protobuf.nuspec]: Were these specifically added by Oren's add-in, or were they hand-written?

Comment (jtattermusch) [inline csharp/src/Google.Protobuf/Google.Protobuf.nuspec]: The dependencies are autogenerated by the NuSpec.ReferenceGenerator.

File: csharp/src/Google.Protobuf/Google.Protobuf.nuspec (hunk 1 above)
Comment (jtattermusch) [inline csharp/src/Google.Protobuf/Google.Protobuf.nuspec]: I am not 100% sure, trying to test it.

Comment (jtattermusch) [inline csharp/src/Google.Protobuf/Google.Protobuf.nuspec]: Just added them, but still not able to make this work with the latest version of 
https://dotnetcli.blob.core.windows.net/dotnet/dev/Installers/Latest/dotnet-win-x64.latest.msi

Comment (jskeet) [inline csharp/src/Google.Protobuf/Google.Protobuf.nuspec]: I wouldn't worry too much about that part yet (much can change) - but I think we do want it in lib/dotnet. LGTM with the changes now listed.

**Comment by jtattermusch** [comment]
Rebased to clean history and to rerun the tests. Feel free to merge once green.

CC @xfxyjwf   this PR needs to be merged into the release branch before creating the tag.

**Comment by jtattermusch** [comment]
Based on what travis says, the referenceGenerator nuget package is causing a build error on Linux (probably when running the tool to update the .nuspec file). One option would be to uninstall the package, (as the .nuspec file has already been updated) but then we would have no way of knowing if the list of dependencies in the nuget package stays up to date in the future.

An ideal option would be to add some kind of "sanity" check that tries to regenerate the dependencies and verifies that the checked-in .nuspec file is the same as the one regenerated. I can file an issue for such sanity check tomorrow. For the upcoming release, let's just submit the .nuspec file and remove the reference generator package from the project.

**Comment by jskeet** [comment]
SGTM. It may be worth pinging Oren to let him know about the issue too.

Jon

**Comment by jtattermusch** [comment]
Updated this PR, created and issue #1054 and upvoted and existing issue on Oren's repo: https://github.com/onovotny/ReferenceGenerator/issues/16

**Comment by jskeet** [comment]
LGTM - merge at will.

**Comment by onovotny** [comment]
Hi,

I've commented on the issue... on the issue thread on my repo.

**Pull request merged**